app/src/main/java/com/hifeful/mealmania/presentation/myMeals/MyMealsViewModel.kt

```diff
@@ -0,0 +1,56 @@
+package com.hifeful.mealmania.presentation.myMeals
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.hifeful.mealmania.domain.repository.MealsRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.consumeAsFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class MyMealsViewModel @Inject constructor(
+    private val mealsRepository: MealsRepository
+) : ViewModel() {
+
+    val events = Channel<MyMealsEvent>(Channel.UNLIMITED)
+    val state = MutableStateFlow(MyMealsState())
+
+    init {
+        handleEvents()
+    }
+
+    private fun handleEvents() {
+        viewModelScope.launch {
+            events.consumeAsFlow()
+                .collect { event ->
+                    when (event) {
+                           is MyMealsEvent.LoadRecentMeals -> mealsRepository.getRecentMeals()
+                            .subscribeOn(Schedulers.computation())
+                            .observeOn(AndroidSchedulers.mainThread())
```

Comment:
When we have Rx and Coroutines together, we should convert Rx into Coroutine.
That way, we gonna have control of threads and easily debug and handle any errors.
👌 The known and checked solution of how to do this conversion is using suspendCoroutine:
```
suspendCoroutine { continuation ->
    try {
          continuation.resume(mealsRepository.getRecentMeals().blockingGet())
    } catch (cancellation: CancellationException) {
          continuation.resumeWithException(cancellation)
    } catch (failure: Exception) {
          // handle error somehow (for instance, map to custom error type)
          continuation.resumeWithException(cancellation)
    }
 }
```

Reply:
No action is required for now
As threading will be in another module soon